pkg/v1/cluster/schemas.go

```diff
@@ -27,6 +27,25 @@ const (
 	StatusUnknown                            Status = "UNKNOWN"
 )
 
+func getSupportedStatuses() []Status {
+	return []Status{
+		StatusActive, StatusPendingCreate, StatusPendingUpdate, StatusPendingUpgrade, StatusPendingRotateCerts,
+		StatusPendingDelete, StatusPendingResize, StatusPendingNodeReinstall, StatusPendingUpgradePatchVersion,
+		StatusPendingUpgradeMinorVersion, StatusPendingUpdateNodegroup, StatusPendingUpgradeMastersConfiguration,
+		StatusPendingUpgradeClusterConfiguration, StatusMaintenance, StatusError, StatusUnknown,
+	}
+}
+
+func isStatusSupported(s Status) bool {
```

Comment:
Not sure but until there are not too many supported statuses iteration over slice to check if status is supported looks ok.
But if there any doubts I can rewrite this function and use `map[Status]bool` to get constant search time.

Unfortunately don't know how nicely reduce code duplication in `getSupportedStatuses` func and consts declaration. But, again imho, until there are not so many statuses it looks fine.